File: recruitment-automation-system/src/main/java/com/RAS/recruitment_automation_system/application/ApplicationRequest.java
package com.RAS.recruitment_automation_system.application;

import com.RAS.recruitment_automation_system.joblisting.JobListing;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;

import java.sql.Date;

public record ApplicationRequest(
        Integer id,

        @NotEmpty(message = "Le nom du candidat ne peut pas être vide")
        String candidateName,

        @NotEmpty(message = "L'email du candidat ne peut pas être vide")
        String candidateEmail,

        @NotEmpty(message = "L'URL du CV ne peut pas être vide")
        String resumeUrl,

        String coverLetter,

        String status,

        Date applicationDate,

        @NotNull(message = "L'ID du job ne peut pas être nul")
        Integer jobId

) {
}

File: recruitment-automation-system-ui/src/app/services/models/application-request.ts
/* tslint:disable */
/* eslint-disable */
import { JobListing } from '../models/job-listing';
export interface ApplicationRequest {
  applicationDate?: string;
  candidateEmail?: string;
  candidateName?: string;
  coverLetter?: string;
  id?: number;
  jobListing?: JobListing;
  jobListingId?: JobListing;
  resumeUrl?: string;
  status?: string;
}
